config/crd/experimental/gateway.networking.k8s.io_gateways.yaml

```diff
@@ -103,6 +103,31 @@ spec:
                 maxLength: 253
                 minLength: 1
                 type: string
+              infrastructure:
+                description: "Infrastructure defines infrastructure level attributes
+                  about this Gateway instance. \n <gateway:experimental>"
+                properties:
+                  routability:
+                    description: "Routability allows the Gateway to specify the accessibility
+                      of its addresses. Setting this property will override the default
+                      value defined by the GatewayClass. \n If the desired Gateway
+                      routability is incompatible with the GatewayClass implementations
+                      MUST set the condition `Accepted` to `False` with `Reason` set
+                      to `UnsupportedRoutability`. \n The default value of routability
+                      is implementation specific and MUST remain consistent for Gateways
+                      with the same gatewayClassName \n Implementations MAY leave
+                      this property unset. The routability MUST appear in the GatewayStatusAddress.
+                      \n Implementations MUST clearly document if they support updates
+                      to this field. The default expectation should be that changes
+                      to this field are not supported unless an implementation specifies
+                      otherwise. \n If a Gateway is mutated but does not support the
+                      desired routability it MUST set the `Accepted` and `Programmed`
+                      conditions to `False` with `Reason` set to `UnsupportedRoutability`."
+                    maxLength: 253
+                    minLength: 1
+                    pattern: ^Public|Private|Cluster|[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*\/[A-Za-z0-9\/\-_]+$
```

Comment:
Can you add some examples that will fail this validation to `hack/invalid-examples` and examples that will pass to `examples/experimental`?

Reply:
done